contrib/pg_tde/src/access/pg_tde_tdemap.c

```diff
@@ -524,6 +524,43 @@ pg_tde_perform_rotate_key(TDEPrincipalKey *principal_key, TDEPrincipalKey *new_p
 	}
 }
 
+void
+pg_tde_delete_principal_key_redo(Oid dbOid)
+{
+	char		path[MAXPGPATH];
+
+	LWLockAcquire(tde_lwlock_enc_keys(), LW_EXCLUSIVE);
+
+	pg_tde_set_db_file_path(dbOid, path);
+	durable_unlink(path, WARNING);
+
+	LWLockRelease(tde_lwlock_enc_keys());
+}
+
+/*
+ * Deletes the principal key for the database. This fucntion checks if key map
+ * file has any entries, and if not, it removes the file. Otherwise raises an error.
+ *
+ * The caller must hold an exclusive lock on the files before calling this function.
```

Comment:
This comment feels a bit pointless with the assert below.

Reply:
Removed